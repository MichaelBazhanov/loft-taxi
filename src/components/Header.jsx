import React from "react";
import logoTaxi from '../assets/images/logo-taxi.svg';
import PropTypes from "prop-types";

const Header = (props) => {
	return (
		<header className="bg-black-me">
			<div className="container mx-auto py-5 px-8">

				<div className="flex justify-between items-center">
					<img src={logoTaxi} className="logo-taxi" alt="logo" />

					<nav className="flex text-white space-x-7">
						<button onClick={() => { props.navigate('map') }} type="button" className="hover:text-yellow-me text-xl">Карта</button>
						<button onClick={() => { props.navigate('profile') }} type="button" className="hover:text-yellow-me text-xl">Профиль</button>
						<button onClick={() => {
							props.logOut()
							props.navigate('login')
						}} type="button" className="hover:text-yellow-me text-xl">Выйти</button>
					</nav>

				</div>
			</div>
		</header>
	)
}
Header.propTypes = {
	props: PropTypes.shape({
		logOut: PropTypes.func.isRequired,
		navigate: PropTypes.func.isRequired
	})
}

export default Header