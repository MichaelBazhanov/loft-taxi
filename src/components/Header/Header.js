import { useEffect, useState } from "react";
import { connect } from "react-redux";
import { logOut } from "../../modules/authorization"; //просто импортируем action
import PropTypes from "prop-types";

import HeaderDesktop from "../HeaderDesktop";
import HeaderMobile from "../HeaderMobile";

const Header = ({ logOut }) => {
  const [width, setWidth] = useState(window.innerWidth);

  const updateDimensions = () => {
    setWidth(window.innerWidth);
  };

  useEffect(() => {
    window.addEventListener("resize", updateDimensions);
    return () => window.removeEventListener("resize", updateDimensions);
  }, []);

  return (
    <>
      {width >= 640 && <HeaderDesktop logOut={logOut} />}
      {width < 640 && <HeaderMobile logOut={logOut} />}
    </>
  );
};
Header.propTypes = {
  props: PropTypes.shape({
    logOut: PropTypes.func.isRequired,
  }),
};

export default connect(null, { logOut })(Header);
