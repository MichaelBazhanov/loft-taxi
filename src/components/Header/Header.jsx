import React from "react";
import './index.css'
import logoTaxi from '../../images/logo-taxi.svg';

const Header = () => {
	return (
		<header className="header">
			<div className="wrapper">
				<div className="header-container">
					<div className="header__logo">
						<img src={logoTaxi} className="logo-taxi" alt="logo" />
					</div>
					<nav className="header__nav-items">
						<a className="header__nav-item" href="#">Карта</a>
						<a className="header__nav-item" href="#">Профиль</a>
						<a className="header__nav-item" href="#">Выйти</a>
					</nav>
				</div>
			</div>
		</header>
	)
}

export default Header