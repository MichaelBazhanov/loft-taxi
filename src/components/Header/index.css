.header {
  width: 100%;
  height: 102px;
	min-height: 102px;
  background: #1c1a19;
	display: flex;
	justify-content: space-around;
}
.logo-taxi {
}

