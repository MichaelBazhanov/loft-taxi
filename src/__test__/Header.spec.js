import React from 'react'
import TestRenderer from 'react-test-renderer'
import { Provider } from 'react-redux'

import Header from '../components/Header'

// Мокаем NavLink что бы внутри <Header /> отдал просто функцию
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  NavLink: () => <div>NavLink component</div>,
}))

// Формируем замоканный store
const mockStore = {
  getState: () => {},
  subscribe: () => {},
  dispatch: () => {
    // диспачим внутри компонента
    logOut: jest.fn()
  },
}

it('Renders correctly', () => {
  const HeaderExample = TestRenderer.create(
    <Provider store={mockStore}>
      <Header />
    </Provider>,
  ).toJSON()

  expect(HeaderExample).toMatchSnapshot()
})
