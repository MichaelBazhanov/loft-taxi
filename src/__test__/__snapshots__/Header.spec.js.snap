// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<header
  className="bg-black-me"
>
  <div
    className="container mx-auto py-5 px-8"
  >
    <div
      className="flex justify-between items-center"
    >
      <img
        alt="logo-taxi"
        src="logo-taxi.svg"
      />
      <nav
        className="flex text-white space-x-7"
      >
        <button
          className="hover:text-yellow-me text-xl"
          type="button"
        >
          <div>
            NavLink component
          </div>
        </button>
        <button
          className="hover:text-yellow-me text-xl"
          type="button"
        >
          <div>
            NavLink component
          </div>
        </button>
        <button
          className="hover:text-yellow-me text-xl"
          onClick={[Function]}
          type="button"
        >
          <div>
            NavLink component
          </div>
        </button>
      </nav>
    </div>
  </div>
</header>
`;
